### PR TITLE
Jetpack connect: Fix automattician approval screen

### PR DESCRIPTION
**Pull request by johnHackworth**
Right now we are redirecting 100% of jetpack auth attempts to the new auth screen for testing when it comes from a user that have a @automattic.com email. 

When a user disconnects a site and the reconnect it, since this screen wasn't updating the calypso's site list, they were getting stuck in the "browse upgrades" button, because calypso thought that the site was already connected. Now, when the connection flow hasn't being started in calypso, we force a reload of the sites list so we have updated info.

This won't be available for any external user, but will stop screwing automatticians trying to test jetpack stuff :) 

image

how to test
=========
1. Connect a site using a 3rd party site user that have an @automattic.com email
2. Disconnect the site
2. Connect it again. You shouldn't get stucked in the false "browse upgrades" button

ping @enejb @dereksmart @lsinger

### CODE REVIEW

**Comment by enejb** [comment]
Cool will check it out soon :) Thanks for such a quick fix!

**Comment by enejb** [comment]
@johnHackworth I am not sure how to test this. What do I need to change in Jetpack to redirect it to the local version that is running your branch?

**Comment by johnHackworth** [comment]
uhm, good point, @enejb you need a jetpack sandbox pointing to you .com sandbox ... and then go to class.jetpack-server-version.php and change `get_calypso_url` to make sure that it always redirect you to calypso.localhost (also, `redirect_auth_to_calypso` should always return true if you are using a test user that hasn't a @automattic.com email)

**Comment by roccotripaldi** [comment]
Rebased and added a few commits. Feel free to squash.

File: client/signup/jetpack-connect/authorize-form.jsx [hunk 1]
```diff
@@ -159,6 +161,10 @@ const LoggedInForm = React.createClass( {
 		const { autoAuthorize, queryObject } = this.props.jetpackConnectAuthorize;
 		debug( 'Checking for auto-auth on mount', autoAuthorize );
 		this.props.recordTracksEvent( 'jpc_auth_view' );
+		if ( ! this.props.isCalypsoStartedConnection ) {
+			debug( 'refetching sites on mount' );
```
Comment (roccotripaldi) [inline client/signup/jetpack-connect/authorize-form.jsx]: I added in this debug. I've noticed that we are requesting sites even on connections that start in Calypso. `isCalypsoStartedConnection` seems to be broken.

**Comment by roccotripaldi** [comment]
I played around a little bit more, and found that a wonky variable name was causing my troubles. All fixed. LGTM!

**Comment by mtias** [comment]
Needs rebase.

**Comment by johnHackworth** [comment]
rebased already... the use case this fixes is not there anymore (you can't access to the new auth screen if you started your flow in jetpack). However, this is the base for the migration from the old auth screen to the new one for this case, and we plan to do this in the next month or so, so it would be nice to merge this changes so we can start building that integration from here.

**Comment by johnHackworth** [comment]
This PR is not longer needed after the last changes in JPC. I'm closing it

**Pull request closed without merging**